File: demo/src/main/java/com/devbrackets/android/recyclerextdemo/database/DBHelper.java
package com.devbrackets.android.recyclerextdemo.database;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import java.util.List;

/**
 *
 */
public class DBHelper extends SQLiteOpenHelper {
    private static final int database_VERSION = 1;
    private static final String database_NAME = "demo";

    public DBHelper(Context context) {
        super(context, database_NAME, null, database_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(ItemDAO.CREATE_TABLE);
        populateItemDAOTable(db);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL(ItemDAO.DROP_TABLE);
        this.onCreate(db);
    }

    private void populateItemDAOTable(SQLiteDatabase database) {
        //Only add items if we haven't already
        List<ItemDAO> items = ItemDAO.findAll(database);
        if (items.size() > 0) {
            database.close();
            return;
        }

        //create and save some dummy items...
        for (int i = 1; i <= 200; i++) {
            ItemDAO item = new ItemDAO("RecyclerExt Demo Item " + i);
            item.save(database);
        }

    }
}

File: demo/src/main/kotlin/com/devbrackets/android/recyclerextdemo/data/database/DBHelper.kt
package com.devbrackets.android.recyclerextdemo.data.database

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

/**
 * A simple utility for interacting with the database
 */
class DBHelper(context: Context?) : SQLiteOpenHelper(context, database_NAME, null, database_VERSION) {
    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(ItemDAO.CREATE_TABLE)
        populateItemDAOTable(db)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL(ItemDAO.DROP_TABLE)
        onCreate(db)
    }

    /**
     * Populates the database with dummy items to be used for any of the
     * cursor, and most of the list examples.
     *
     * @param database The database to populate
     */
    private fun populateItemDAOTable(database: SQLiteDatabase) {
        //Only add items if we haven't already
        val items: List<ItemDAO?> = ItemDAO.findAll(database)
        if (items.isNotEmpty()) {
            database.close()
            return
        }

        //create and save some dummy items...
        for (i in 1..500) {
            val item = ItemDAO("RecyclerExt Demo Item $i")
            item.save(database)
        }
    }

    companion object {
        private const val database_VERSION = 1
        private const val database_NAME = "demo"
    }
}